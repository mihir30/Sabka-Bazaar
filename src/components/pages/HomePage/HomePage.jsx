import React, { useEffect, useState } from "react";
import { Carousel, Section } from "../../organisms";
import { getData } from "../../utils/common.utils";
import { WithSpinner } from "../../atoms";
const HomePage = () => {
  const [category_data, setData] = useState([]);
  useEffect(() => {
    const fetchData = async () => {
      try {
        const categories = await getData("http://localhost:5000/categories");
        categories.sort((category1, category2) => {
          return category1.order - category2.order;
        });
        setData(categories);
      } catch (error) {
        console.log(error);
      }
    };
    fetchData();
  }, []);
  return (
    <div aria-label="homepage-container">
      <div aria-label="homepage-carousel">
        <Carousel />
      </div>
      <div aria-label="homepage-categories">
        {
          category_data && category_data.length > 0 ?
            category_data.map((data, index) => {
              return (
                data.enabled && (
                  <Section
                    imgSrc={data.imageUrl}
                    imgAlt="image"
                    key={index}
                    id={data.id}
                    name={data.name}
                    value={data.key}
                    description={data.description}
                    order={data.order}
                  />
                )
              );
            }) : <WithSpinner />
        }
      </div>
    </div>
  );
};

export default HomePage;
