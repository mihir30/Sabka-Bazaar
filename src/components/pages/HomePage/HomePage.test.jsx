import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import HomePage from "./HomePage";
import renderer from 'react-test-renderer';

describe("checking the category labels are rendered properly",()=>{
    it('bakery label',async()=>{
        render(<BrowserRouter><HomePage/></BrowserRouter>);
        const label=await screen.findByText('Bakery Cakes and Dairy');
        expect(label).toBeInTheDocument();
    });
    it('beverages label',async()=>{
        render(<BrowserRouter><HomePage/></BrowserRouter>);
        const label=await screen.findByText('Beverages');
        expect(label).toBeInTheDocument();
    });
    it('baby care label',async()=>{
        render(<BrowserRouter><HomePage/></BrowserRouter>);
        const label=await screen.findByText('Baby Care');
        expect(label).toBeInTheDocument();
    });
    it('beauty and hygiene',async()=>{
        render(<BrowserRouter><HomePage/></BrowserRouter>);
        const label=await screen.findByText('Beauty and Hygiene');
        expect(label).toBeInTheDocument();
    });
    it('fruits and vegetables',async()=>{
        render(<BrowserRouter><HomePage/></BrowserRouter>);
        const label=await screen.findByText('Fruits & Vegetables');
        expect(label).toBeInTheDocument();
    });
})
describe("category component",()=>{
    
    it('bakery description', async() => {
        render(<BrowserRouter><HomePage/></BrowserRouter>);
        const categoryText = await screen.findByText('The best cupcakes, cookies, cakes, pies, cheesecakes, fresh bread, biscotti, muffins, bagels, fresh coffee, milk and more.');
        expect(categoryText).toBeInTheDocument();
    });
    it('Beverages description', async() => {
        render(<BrowserRouter><HomePage/></BrowserRouter>);
        const categoryText = await screen.findByText('Our beverage department will ensure your fridge is always fully stocked. Shop for soda, juice, beer and more.');
        expect(categoryText).toBeInTheDocument();
    });
    
    it('Beauty & hygiene description', async() => {
        render(<BrowserRouter><HomePage/></BrowserRouter>);
        const categoryText = await screen.findByText('Buy beauty and personal care products online in India at best prices.');
        expect(categoryText).toBeInTheDocument();
    });

    
    it('Baby care description', async() => {
        render(<BrowserRouter><HomePage/></BrowserRouter>);
        const categoryText = await screen.findByText('Shop online for Baby Products, Diapers, Skin Care Products,etc.');
        expect(categoryText).toBeInTheDocument();
    });
    

    it('should have right Category component snapshot', () => {
        const tree =renderer.create(<BrowserRouter><HomePage/></BrowserRouter>).toJSON();
        expect(tree).toMatchSnapshot();
    });
})
describe("checking the carousel section images are rendered properly",()=>{
    it('image1',async()=>{
        render(<BrowserRouter><HomePage/></BrowserRouter>);
        const altText=await screen.findByAltText('Independence Day Deal - 25% off on shampoo');
        expect(altText).toBeInTheDocument();
    });
})