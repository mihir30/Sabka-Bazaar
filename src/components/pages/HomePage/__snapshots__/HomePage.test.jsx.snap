// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`category component should have right Category component snapshot 1`] = `
<div
  aria-label="homepage-container"
>
  <div
    aria-label="homepage-carousel"
  >
    <section
      className="carousel-section"
    >
      <div
        className="spinner-overlay"
      >
        <div
          className="spinner-container"
        />
      </div>
      <button
        className="carousel-prev-button carousel-button"
        onClick={[Function]}
        type="button"
      >
        PREV
      </button>
      <button
        className="carousel-next-button carousel-button"
        onClick={[Function]}
        type="button"
      >
        NEXT
      </button>
      <div
        className="banner-dots-container desktop"
      />
    </section>
    <div
      className="banner-dots-container mobile"
    />
  </div>
  <div
    aria-label="homepage-categories"
  >
    <div
      className="spinner-overlay"
    >
      <div
        className="spinner-container"
      />
    </div>
  </div>
</div>
`;
